src/v/cloud_roles/refresh_credentials.cc

```diff
@@ -195,6 +202,8 @@ ss::future<> refresh_credentials::fetch_and_update_credentials() {
       });
 }
 
+ss::future<> refresh_credentials::stop() { co_await _gate.close(); }
```

Comment:
There's a sleep in our now-gated credentials-fetching loop, which IIUC appears to cap out at 5 minutes. Let's hit the abort source here too so that sleep gets terminated

Reply:
good point, fixed